Technical release notes for `release-2023-09` (#725)
livingdocsIO/documentation: content/operations/releases/release-2023-09.md

livingdocs-server migrate up
```

### Drop support for AngularJS metadata plugins 🔥

Custom AngularJS metadata plugins are no longer supported. You should migrate to core metadata plugins, full list [here](https://docs.livingdocs.io/reference/document/metadata/plugins/). If you cannot replace AngularJS plugins with core plugins: contact your Livingdocs customer manager before you write new plugins in Vue.

This includes Angular Metadata UI components, Metadata Services, mixing custom UI with Livingdocs core metadata types and custom types with Livingdocs UI components.

> 

Author: Do we really want customers to migrate from their angular plugins to vue without talking to us first? My impression is that we want to help them solve the missing functionality on the core. The migration is an easy way out to not even consider upstream plugins. Do you think we should include the link regardless?

> 

Reviewer: Every help is welcome and in the migration path of the provided link there is also a mention that they should talk with us.

### Drop support for `li-media-language` metadata plugin 🔥

Metadata plugin type `li-media-language` got replaced with `li-metadata-translations`. Upgrade path using `li-metadata-translations` instead which supports the same config as `li-media-language`.

* [Remove `li-media-language` metadata plugin](https://github.com/livingdocsIO/livingdocs-server/pull/6054)

### Preview API 🔥

The previewApi (`li-preview`) feature support has been removed. Use `registerPreviewFunction` instead. See details [here](https://docs.livingdocs.io/customising/server-configuration/#custom-previews).

* [Remove `li-preview` feature](https://github.com/livingdocsIO/livingdocs-server/pull/6008)

## Deprecations

### `document.customPublicationDateField` is deprecated

Using editor config `document.customPublicationDateField` is deprecated and will be removed with `release-2023-11`. Please migrate to `contentType.publishControl.publishSchedule`.

* [Remove `document.customPublicationDateField` config](https://github.com/livingdocsIO/livingdocs-editor/pull/7259)

### `li-reference` metadata plugin is deprecated

> Metadata plugin type `li-reference` has been deprecated. Please replace it with `li-document-reference`. No data migration is required, but you will need to remove `referenceType` from the metadata plugin config.

Reviewer: for deprecations: always provide a release, when the break happens (if it's unknown, then it's also good to mention that)

```
Notice the Ticker Host contentType has a `ticker` property, which defines the Ticker Entry contentType. The Ticker Entry contentType has a metadata property `tickerHost`, which defines the reference to the ticker host. This reference is set on creation of each ticker entry.

For more information and limitations of the Ticker Feature, see [here]().

### Anchor Linking

When selecting a document in the inline link edit tool or in `li-tree`, it is now possible to also select a specific component within that document. The directiveId (= componentId:directiveName) will then be included in the URL hash part.

To enable anchor linking for a project, at least one component with a doc-editable and `anchor: true` must be defined within the project config. For example:

```
  name: 'title',
  label: 'Title',
  directives: [{
    type: 'editable',
    name: 'title',
    anchor: true
  }],
  html: dedent`
    <h2 doc-editable="title">
      Title
    </h2>
  `
```

### Display and Base Filters

In the July 2023 release, we introduced in the possibility for users to create configurable filters. With this release, we have added filter support for more core metadata plugins.

> We have added the possibility to configure display filters for the following metadata plugins:

Reviewer: I'm thinking about a feature for our documentation repo. As soon as you mention a Livingdocs entity like `Display Filter`, Hugo automatically creates a link to the guide or reference doc based on a manually defined mapping.

What do you think about that @peyerluk @meinradandermatt ?

- `li-string-list` with `dataProvider`
- `li-enum` with `dataProvider`
- `li-text` with `dataProvider`
- `li-document-reference` with `style: minimal`
- `li-document-references` with `style: minimal`

We have added the possibility to configure base filters for the following metadata plugins:
- `li-desknet-integration`
- `li-retresco`
- `li-imatrics-nlp-tags`

> For these filters to take effect you must index the metadata properties in Elasticsearch/Opensearch. To do so you have to define `index: true` in the metadata plugin config. See [here](https://docs.livingdocs.io/reference/document/metadata/plugins/#indexing) for more information. Then, reindex the document using:

Reviewer: I think we should start improving the Project Config validation (this is just one case, but we have a lot of such dependencies). If you configure a core display filter but forget to add to configure the indexing, people could be confused. I guess such cases lead to support questions. 
fyi @marcbachmann

Administrators can define which languages are available in the language dropdown by configuring `app.availableLocales` in the editor configuration. Supported languages are `en`, `de`, `fr` and `it`. The language switch stays hidden if `app.availableLocales` is `undefined` or `[]`.

```js
app: {
  locale: 'en',
  availableLocales: ['en', 'de', 'fr', 'it']
}
```

### Publish Control UX improvements

We have improved Publish Control UI to make it easier to use. No configuration changes are required from developers perspective to take advange of the improvements. The main improvements are:
- The visual feedback for Metadata errors blocking a publish is improved
- Unpublish Buttons use red text color now to be more clear about the potentially damaging effects
- The confirm step presented when a publish is scheduled and the user clicks "Publish Now" has more visual clarity now

### Cross Project Content Sharing

> 

Author: We discussed with Beni that we wanted to clearly define what are the feature limitations. Just to make sure other customers don't believe it will be able to do more than it can do. That is why we added it here.

* [Remove `document.customPublicationDateField` config](https://github.com/livingdocsIO/livingdocs-editor/pull/7259)

### `li-reference` metadata plugin is deprecated

Metadata plugin type `li-reference` has been deprecated. Please replace it with `li-document-reference`. No data migration is required, but you will need to remove `referenceType` from the metadata plugin config.

* [Reference plugin feature parity](https://github.com/livingdocsIO/livingdocs-server/pull/6030)

### `li-reference-list` metadata plugin is deprecated

Metadata plugin type `li-reference-list` has been deprecated. Please replace it with `li-document-references`. No data migration is required, but you will need to remove `referenceType` from the metadata plugin config.

* [Reference plugin feature parity](https://github.com/livingdocsIO/livingdocs-server/pull/6030)

### `storageFormat.label` property within the `li-language` metadata plugin is deprecated

The `storageFormat.label` property within the `li-language` metadata plugin is deprecated and will be removed in `release-2024-01`.

## Features

### Ticker Tool

> Livingdocs now offers a tool to create ticker articles with ease. This feature uses a contentType as Ticker Host document, and another contentType as the Ticker Entry. The Ticker Host is used to create a new ticker article, and the Ticker Entry is used to create new entries in the ticker. The Ticker Host can be configured to allow only specific Ticker Entries, and the Ticker Entry can be configured to allow only specific Ticker Hosts.

Reviewer 2: ```suggestion
Livingdocs now offers a tool to create ticker articles with ease. This feature uses a contentType as Ticker Host document, and another contentType as the Ticker Entry. The Ticker Host is used to create a new ticker article, and the Ticker Entry is used to create new entries in the ticker. 
```

An example Ticker Host contentType configuration:

> ```

Reviewer 2: In general in the documentation we can try to make examples as minimal as possible so the reader can focus on the feature at hand.

I'll try:
```
handle: 'tickerHost',
documentType: 'article',
info: {
  label: 'Ticker'
},

ticker: {
  // The tickerEntry contentType
  entryContentType: 'tickerEntry'
},

// ...the rest of your content type configuration
// like metadata, components, publishControl etc.
```

Ticker entry contentType configuration. Note that it does not contain anything ticker specific. The `ticker` config in the tickerHost provides all necessary info to Livingdocs.
```
handle: 'tickerEntry',

// ...the rest of your content type configuration
// like metadata, components, publishControl etc.
```

  {name: 'image-named-crops'},
  {name: 'video'},
  {name: 'video-include'},
  {name: 'teaser-include'},
  {name: 'teaser-include-small'},
  {name: 'teaser-include-translated'},
  {name: 'teaser-list-include'},
  {name: 'teaser-list-include-params-schema'},
  {name: 'insta'},
  {name: 'echo'},
  {name: 'youtube'},
  {name: 'twitch'},
  {name: 'social'},
  {name: 'oembed-include'},
  {name: 'free-html'},
  {name: 'container'}
],

metadata: [
  {
    handle: 'title',
    type: 'li-text',
    ui: {
      label: {
        en: 'Title',
        de: 'Titel'
      }
    }
  }
],

editorWrapper: '<section class="container doc-section"></section>',
defaultContent: [
  {component: 'title'}
]
```
An example Ticker Entry contentType configuration:
```
handle: 'tickerEntry',
documentType: 'article',
info: {
  label: 'Ticker Entry'
},

// Content
editorWrapper: '<section class="container doc-section" style="padding-bottom: 20px;"></section>',
defaultContent: [
  {
    component: 'title'
  }
],

components: [
  {name: 'title'},
  {name: 'p'}
],

// Metadata
metadata: [
  {
    handle: 'tickerHost',
    type: 'li-document-reference',
    config: {
      contentType: 'tickerHost'
    }
  }
]
```

> Notice the Ticker Host contentType has a `ticker` property, which defines the Ticker Entry contentType. The Ticker Entry contentType has a `metadata` property, which defines the Ticker Host contentType. This is how the Ticker Host and Ticker Entry are linked together.

Reviewer 2: ```suggestion
Notice the Ticker Host contentType has a `ticker` property, which defines the Ticker Entry contentType. The Ticker Entry contentType has a metadata property `tickerHost`, which defines the reference to the ticker host. This reference is set on creation of each ticker entry.
```

> For more information and limitations of the Ticker Tool, see [here]().

Reviewer 2: ```suggestion
For more information and limitations of the Ticker Feature, see [here]().
```

### Multilanguage includes in French and Italian

Livingdocs UI has been translated to more languages. We now support French and Italian on top of English and German. A new language switcher has been added to the user profile for a personalized experience. By default, the language defined in the editor configuration `app.locale` is used.

Administrators can define which languages are available in the language dropdown by configuring `app.availableLocales` in the editor configuration. Supported languages are `en`, `de`, `fr` and `it`. The language switch stays hidden if `app.availableLocales` is `undefined` or `[]`.

```js
app: {
  locale: 'en',
  availableLocales: ['en', 'de', 'fr', 'it']
}
```

### Publish Control UX improvements

We have improved Publish Control UI to make it easier to use. No configuration changes are required from developers perspective to take advange of the improvements. The main improvements are:
- The visual feedback for Metadata errors blocking a publish is improved
- Unpublish Buttons use red text color now to be more clear about the potentially damaging effects

> - The confirm step presented when a publish is scheduled and the user clicks "Publish Now" has more visual clarity now

Reviewer 3: We could remove this from the technical release notes if there are no actions required by a developer.

Author: I was thinking about that while writing them, but Beni said we should include  them. @ivester do you remember what Beni wanted to highlight there?

Reviewer 4: I remember Beni pointing this out but I can't remember why it is relevant for the tech release notes. I guess it doesn't hurt to leave it here.
